code/modules/mob/living/carbon/xenomorph/Abilities.dm

```diff
@@ -354,47 +354,121 @@
+			if(istype(path, /datum/action/xeno_action/pheromones))
+				path.remove_action(X)
+	else
+		PheromonesOpen = TRUE
+		to_chat(X, "<span class ='xenonotice'>You open the pheromone button choices.</span>")
+		var/list/subtypeactions = subtypesof(/datum/action/xeno_action/pheromones)
+		for(var/path in subtypeactions)
+			var/datum/action/xeno_action/pheromones/A = new path()
+			A.give_action(X)
+
+/datum/action/xeno_action/pheromones
+	name = "SHOULD NOT EXIST"
+	plasma_cost = 30 //Base plasma cost for begin to emit pheromones
+
+/datum/action/xeno_action/pheromones/can_use_action()
+	var/mob/living/carbon/Xenomorph/X = owner
+	if(X.is_mob_incapacitated() || X.lying || X.buckled)
+		return FALSE
+	return TRUE
 
-/datum/action/xeno_action/emit_pheromones/action_activate()
+/datum/action/xeno_action/pheromones/action_activate() //Must pass the basic plasma cost; reduces copy pasta
 	var/mob/living/carbon/Xenomorph/X = owner
-	if(!X.check_state())
-		return
+	if(!X.check_plasma(plasma_cost))
+		to_chat(X, "<span class='xenowarning'>You need more than [plasma_cost] to emit this pheromone.</span>")
+		return FALSE
+	return TRUE
 
-	if(X.current_aura)
+/datum/action/xeno_action/pheromones/emit_recovery //Type casted for easy removal/adding
+	name = "Emit Recovery Pheromones (30)"
+	action_icon_state = "emit_recovery"
+
+/datum/action/xeno_action/pheromones/emit_recovery/action_activate()
+	var/mob/living/carbon/Xenomorph/X = owner
+	if(X.current_aura == "recovery")
+		X.visible_message("<span class='xenowarning'>\The [X] stops emitting strange pheromones.</span>", \
+		"<span class='xenowarning'>You stop emitting [X.current_aura] pheromones.</span>", null, 5)
 		X.current_aura = null
-		X.visible_message("<span class='xenowarning'>\The [X] stops emitting pheromones.</span>", \
-		"<span class='xenowarning'>You stop emitting pheromones.</span>", null, 5)
 	else
-		if(!X.check_plasma(30))
-			return
-		var/choice = input(X, "Choose a pheromone") in X.xeno_caste.aura_allowed + "help" + "cancel"
-		if(choice == "help")
-			to_chat(X, "<span class='notice'><br>Pheromones provide a buff to all Xenos in range at the cost of some stored plasma every second. Burning Xenos can neither emit nor benefit from pheromones. Effects are as follows:<br><B>Frenzy</B> - Increased run speed, damage and tackle chance.<br><B>Warding</B> - Increased armor, reduced incoming damage and critical bleedout.<br><B>Recovery</B> - Increased plasma and health regeneration.<br></span>")
-			return
-		if(choice == "cancel") return
-		if(!X.check_state()) return
-		if(X.current_aura) //If they are stacking windows, disable all input
-			return
-		if(!X.check_plasma(30))
-			return
-		X.use_plasma(30)
-		X.current_aura = choice
-		X.visible_message("<span class='xenowarning'>\The [X] begins to emit strange-smelling pheromones.</span>", \
-		"<span class='xenowarning'>You begin to emit '[choice]' pheromones.</span>", null, 5)
-		playsound(X.loc, "alien_drool", 25)
+		. = ..()
+		if(.)
```

Comment:
What is being checked here could be moved to a proc not unlike `can_use_action()`.